Project1/GridCell.cpp

```diff
@@ -0,0 +1,63 @@
+#include "GridCell.h"
+
+GridCell::GridCell(): _tile() {
+}
+
+CellEdge GridCell::getEdge(GridCell* other_cell) {
```

Comment:
Your getter has side effects...

This should return std::optional<CellEdge>. Then you can check whether it was found by calling hasValue on the return value.